init.lua

```diff
@@ -89,19 +89,51 @@ function torch.type(obj)
 end
 
 --[[ See if a given object is an instance of the provided torch class. ]]
+local function exactTypeMatch(obj_mt, typeSpecNameString)
+  return obj_mt.__typename == typeSpecNameString
+end
+
+local function partialTypeMatch(obj_mt, typeSpecNameString)
```

Comment:
The problem with this approach is that, in its current form, isTypeOf allows the usage of wildcards. For example, if `tensor` is a `FloatTensor`, then `torch.isTypeOf(tensor, "*Tensor")` returns true. This functionality is not provided by the solution above, nor by mahaeb@'s solution.

I am very much in favor of removing the wildcard functionality altogether, but if we do, we should first try and understand whether and how this is actually used, because this can break existing code. The wildcards were originally used in `torch.isTensor`, but this usage has been removed in a [recent commit](https://github.com/torch/torch7/commit/bf269f0f9bc49dd68cf37a0504e2eb70688a7a85).